index.js

```diff
@@ -142,7 +227,24 @@ function getAllMoviesReleasedAtOrBeforeYear() {}
  *  getBiggestBoxOfficeMovie(movies);
  *  //> "Incredibles 2"
  */
-function getBiggestBoxOfficeMovie() {}
+function getBiggestBoxOfficeMovie(movies) {
+  if (movies.length === 0){
+    return null;
+  }
+  let movieName = "";
+  let highestBox = Number((((movies[0].boxOffice).slice(1,movies[0].boxOffice.length)).replace(",","")).replace(",",""));
+  for (let i = 0; i < movies.length; i++){
+    if ((Number((((movies[i].boxOffice).slice(1,movies[i].boxOffice.length)).replace(",","")).replace(",",""))) > highestBox){
+      highestBox = Number((((movies[i].boxOffice).slice(1,movies[i].boxOffice.length)).replace(",","")).replace(",",""));
+    }
+  }
+  for (let i = 0; i < movies.length; i++){
+    if (Number((((movies[i].boxOffice).slice(1,movies[i].boxOffice.length)).replace(",","")).replace(",","")) === highestBox){
+      movieName = movies[i].title;
+    }
+  }
+  return movieName;
+}
```

Comment:
Same comment as above.  One of our best principles in writing clean code is keeping code DRY - which stands for `Don't Repeat Yourself`.  
In this case `movies[i].boxOffice` is the target of many calls -`length`, `.slice` - we could shorten this if we use a variable
Also take a look at building a `regex` this pattern `/\$|,/g`   is a regular expression that will search for any `$` or `,`.  See if you can use that pattern in your `.replace()` call and only call it once.

Reply:
You are absolutely right Tim. I was so nervous because of the time that I had left and I just wrote my code waiting to pass the tests but I glad you are telling me all of the things that I can work on. I am going to rewrite my code and put more focus on what you told me about. Thank you so much for taking the time to review my work and giving me your points of view, I really appreciate it.